includes/controllers/class-message.php

```diff
@@ -67,6 +74,95 @@ public function __construct( $args = [] ) {
 		parent::__construct( $args );
 	}
 
+	/**
+	 * Fetches messages.
+	 *
+	 * @return WP_Rest_Response
+	 */
+	public function fetch_messages() {
+
+		// Check authentication.
+		if ( ! is_user_logged_in() ) {
+			return hp\rest_error( 401 );
+		}
+
+		// Get thread.
+		$thread_ids = hivepress()->request->get_context( 'message_thread_ids' );
```

Comment:
If I remember correctly request context doesn't work in REST, only in front-end templates, or it's ok when testing?

Reply:
It's ok